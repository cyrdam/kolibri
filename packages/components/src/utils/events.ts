// TODO: Should be synchronized with enums/events.ts
export type KoliBriEventType =
	| 'blur'
	| 'change'
	| 'changePage'
	| 'changePageSize'
	| 'click'
	| 'close'
	| 'focus'
	| 'input'
	| 'mousedown'
	| 'reset'
	| 'selection-change' // todo remove
	| 'selectionChange'
	| 'sort'
	| 'submit'
	| 'toggle';

export function stopPropagation(event: Event): void {
	event.stopImmediatePropagation();
	event.stopPropagation();
}

function createKoliBriEvent<T>(type: KoliBriEventType, detail: T): CustomEvent {
	const event = new CustomEvent(`kol-${type}`, {
		bubbles: true,
		cancelable: true,
		composed: true,
		detail: detail,
	});
	return event;
}

function dispatchKoliBriEvent<T>(target: EventTarget, type: KoliBriEventType, detail?: T): boolean {
	const dispatch = target.dispatchEvent(createKoliBriEvent(type, detail));
	return dispatch;
}

/**
 * @deprecated
 */
export function tryToDispatchKoliBriEvent<T>(type: KoliBriEventType, target?: EventTarget, detail?: T): void {
	target && dispatchKoliBriEvent(target, type, detail);
}

export function dispatchDomEvent(target: HTMLElement, type: KoliBriEventType) {
	target.dispatchEvent(new Event(type, { bubbles: true, composed: true }));
}

export function preventDefaultAndStopPropagation(event: Event) {
	event.preventDefault();
	stopPropagation(event);
}
