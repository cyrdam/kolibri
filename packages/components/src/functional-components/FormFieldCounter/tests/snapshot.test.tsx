import { h } from '@stencil/core';
import KolFormFieldCounterFc from '../FormFieldCounter';
import { renderFunctionalComponentToSpecPage } from '../../../utils/testing';

describe('KolFormFieldCounterFc', () => {
	it('should render correctly', async () => {
		const currentLength = 5;
		const maxLength = 10;
		const classNames = 'custom-class';
		const page = await renderFunctionalComponentToSpecPage(() => (
			<KolFormFieldCounterFc currentLength={currentLength} maxLength={maxLength} class={classNames} />
		));

		expect(page.root).toMatchSnapshot();
	});

	it('should render currentLength and maxLength correctly', async () => {
		const currentLength = 5;
		const maxLength = 10;
		const page = await renderFunctionalComponentToSpecPage(() => <KolFormFieldCounterFc currentLength={currentLength} maxLength={maxLength} />);

		expect(page.root?.textContent).toContain(currentLength.toString());
		expect(page.root?.textContent).toContain(maxLength.toString());
	});

	it('should set the correct class names', async () => {
		const classNames = 'custom-class';
		const page = await renderFunctionalComponentToSpecPage(() => <KolFormFieldCounterFc class={classNames} />);

		expect(page.root?.className).toContain('counter');
		expect(page.root?.className).toContain(classNames);
	});
});
