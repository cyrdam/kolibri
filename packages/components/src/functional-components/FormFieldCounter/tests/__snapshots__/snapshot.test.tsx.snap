// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KolFormFieldCounterFc should render correctly 1`] = `
<span aria-atomic="true" aria-live="polite" class="custom-class kol-form-field__counter" data-testid="input-counter">
  5
  <span aria-label="kol-of" role="img">
    /
  </span>
  10
  <span>
    kol-characters
  </span>
</span>
`;
